modules/broker-core/src/test/java/io/ballerina/messaging/broker/core/selector/AndOperationExpressionTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package io.ballerina.messaging.broker.core.selector;
+
+import io.ballerina.messaging.broker.core.Metadata;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class AndOperationExpressionTest {
+    private Metadata metadata =  new Metadata("queue1", "amq.topic", 0);
+    private ConstantExpression c1 = ConstantExpression.createFromNumericDecimal("89");
```

Comment:
Shall we use a data provider for this type of tests? So that we can get add new test scenarios easily.